src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.MovieDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
for the non-read operation, we should use try-finally block checklist says

Reply:
fixed